### DISCUSSION TITLE
Prefetching Relationship Data and Other Data

### DISCUSSION THREAD

**Discussion by mmalessa80** [Budibase/budibase#10327] [Help]
I use MySQL as my datasources for a test application (proof of concept). Although the side panel feature is very handy under certain circumstances, I still find myself requiring navigation to a detail page, which results into the detail page fetching the data and reloading it. Furthermore, any relationship fields or subtables that show related records don't get fetched until the 'parent' record is selected in some way, which is also true for side panel detail views.

This is likely a feature request unless someone can guide me how to do this. What I would like to get accomplished is if the 'home' page loads several tables, related data is fetched synchronously and is stored as part of a data object. This dataobject would then be part of the application itself and would be accessible on other pages simply via indexing, rather than constant reloading. Maybe I am misunderstanding the inner workings of Budibase, but looking at network traffic it certainly seems that there are data loads happening during navigations rather than just selecting a row at an index that is already in a data object.

**Comment by melohagan** [comment]
Hey @mmalessa80 

Can you do this with actions within the *On Screen Load* event?

So something like:

1. **Execute Query** -> Get all the data you want from the various tables with a custom query
2. **Update State** -> Persist the response of the query into App State. This data can be accessed in later screens without further loading.

Relevant docs:
 - [Custom queries](https://docs.budibase.com/docs/data-sources#custom-queries)
 - [App state](https://docs.budibase.com/docs/app-state)
 - [Chaining actions](https://docs.budibase.com/docs/chaining-actions)
